python_modules/dagster/dagster/_core/definitions/repository_definition/repository_data_builder.py

```diff
@@ -417,7 +439,7 @@ def build_caching_repository_data_from_dict(
     )
 
 
-def _process_and_validate_target(
+def _process_and_validate_target_job(
```

Comment:
not your fault but this function is brutal

Reply:
Yes the mutating of the passed in dicts is kinda hard to follow. Perhaps worth a refactor in a follow-up.